### PR TITLE
New icon sources for an item

### PR DESCRIPTION
**Pull request by lolodomo**
Related to openhab/openhab-core#3052

Depends on the merge of PR openhab/openhab-core#3539

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: configuration/items.md [hunk 1]
```diff
@@ -310,14 +312,49 @@ The following guidelines apply to user-added icon files:
...
+Some user interfaces (e.g. Basic UI) don't have this setting because they support both formats transparently.
 Please check the user interface documentation if in doubt.
 Note that image files with the wrong file ending will be ignored.
 
 Users may substitute their own icon for an icon from the default icon set by placing a file in the `$OPENHAB_CONF/icons/classic/` folder with the same filename as the name of the icon being substituted.
 
+#### Icon sources
+
+Some user interfaces supports other icons than those provided by openHAB.
+Thus the most generic reference to an icon is composed of 3 segments separated by a semicolon:
```
Comment (mueller-ma) [inline configuration/items.md:327]:
```suggestion
Thus the most generic reference to an icon is composed of 3 segments separated by a colon:
```

**Review by mueller-ma — COMMENTED** [review]
I'd mention the addon https://community.openhab.org/t/iconify-icon-provider/144508 here.

With it you can have several icon sets that are supported by at least the Android app as well and don't require an internet connection.

**Comment by mueller-ma** [comment]
This PR will close #2027.

File: configuration/items.md [hunk 2]
```diff
@@ -310,14 +312,49 @@ The following guidelines apply to user-added icon files:
 
 **Bitmaps or Vector Graphics:**
 openHAB can work with either Bitmap (`png`) or Vector (`svg`) icon files.
-The format should match the display capabilities of the user interfaces in use (e.g. Basic UI).
+The format should match the display capabilities of the user interfaces in use.
 It is thereby important to decide on one format beforehand; vector graphics are recommended.
 The setting can be changed via UI for most user interfaces.
+Some user interfaces (e.g. Basic UI) don't have this setting because they support both formats transparently.
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply configuration/items.md:318]: iOS also: https://github.com/openhab/openhab-ios/blob/8c9bd276d3b257960ec4992ee7ac530bdc35bd51/OpenHABCore/Sources/OpenHABCore/Util/Endpoint.swift#L162-L163

**Comment by J-N-K** [comment]
Please also add an example how this should look like (for the different sets/options).

File: configuration/items.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lolodomo) [reply configuration/items.md:318]: Changed

**Comment by lolodomo** [comment]
> I'd mention the addon https://community.openhab.org/t/iconify-icon-provider/144508 here.

Added.

**Comment by lolodomo** [comment]
> Please also add an example how this should look like (for the different sets/options).

I added examples to show the definition of items using different icon sources.

**Comment by lolodomo** [comment]
I think this is now ready for a review.

The case of dynamic icons and hyphens in icon name will be updated later in a separate PR in case a change is really implemented in the core framework.

File: configuration/items.md [hunk 3]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
 Users may substitute their own icon for an icon from the default icon set by placing a file in the `$OPENHAB_CONF/icons/classic/` folder with the same filename as the name of the icon being substituted.
 
+#### Icon sources
+
+Some user interfaces supports other icons than those provided by openHAB.
+Thus the most generic reference to an icon is composed of 3 segments separated by a colon:
+
+1. The first segment is the icon source, e.g. "oh"
+
+2. The second segment is the icon set from this source, e.g. "classic"
```
Comment (stefan-hoehn) [inline configuration/items.md:332]: change 2. to 1. (this is why the linter complains)

File: configuration/items.md [hunk 4]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+
+3. The third segment is the name of an icon in this set, e.g. "switch"
```
Comment (stefan-hoehn) [inline configuration/items.md:334]: same like above

**Review by stefan-hoehn — CHANGES_REQUESTED** [review]

File: configuration/items.md [hunk 5]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+
+If the value contains only two segments, the first segment is assumed to be the icon source and the second the icon name.
+In this case, the icon set is to "classic" by default.
+This is used in particular when the icon source only provides a single set of icons.
+As an example, "&lt;material:favorite&gt;" references the "favorite" icon from the Material icons.
```
Comment (stefan-hoehn) [inline configuration/items.md:339]: How about using the following instead? (and then similarly for the others below)

As an example, `<material:favorite>` references the "favorite" icon from the Material icons.

File: configuration/items.md (hunk 3 above)
Comment (lolodomo) [reply configuration/items.md:332]: What do you mean?

File: configuration/items.md [hunk 6]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+| `if` or `iconify` | iconify icons                         | Supported  | Supported but needs to be enabled | Not yet supported | Not yet supported |
+
+You can find a list of [all Material icons here](https://fonts.google.com/icons?icon.set=Material+Icons).
+
+You can find a list of [all Framework7 icons here](https://framework7.io/icons/).
+
+You can find a list of [all iconify icons here](https://icon-sets.iconify.design/).
+Please note that the iconify option requires Internet connectivity on the client to access the external API.
+The WEB browser will cache the retrieved icons to limit the requests and speed up the rendering.
+Certain user interfaces provide a setting to enable this option, e.g. Basic UI disables this option by default but allows it to be enabled.
```
Comment (stefan-hoehn) [inline configuration/items.md:372]: Wouldn't it make sense to provide a link where it is documented how to do so?

File: configuration/items.md [hunk 7]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+
 #### Dynamic Icons
 
-Some icons are dynamically selected by openHAB depending on the Item's state.
+When and only when the icon source is openHAB, some icons are dynamically selected by openHAB depending on the Item's state.
```
Comment (stefan-hoehn) [inline configuration/items.md:376]: I think "if" is more applicable, isn't it?

**Review by stefan-hoehn — CHANGES_REQUESTED** [review]

**Comment by stefan-hoehn** [comment]
Besides my comments, @lolodomo , I really appreciate that you took the time to write about the topic so comprehensively 🙏

File: configuration/items.md (hunk 3 above)
Comment (lolodomo) [reply configuration/items.md:332]: Ok I understood.

File: configuration/items.md (hunk 7 above)
Comment (lolodomo) [reply configuration/items.md]: Changed

File: configuration/items.md [hunk 8]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+But the setting can be changed via UI in certain user interfaces.
 Please check the user interface documentation if in doubt.
 Note that image files with the wrong file ending will be ignored.
 
 Users may substitute their own icon for an icon from the default icon set by placing a file in the `$OPENHAB_CONF/icons/classic/` folder with the same filename as the name of the icon being substituted.
 
+#### Icon sources
+
+Some user interfaces supports other icons than those provided by openHAB.
+Thus the most generic reference to an icon is composed of 3 segments separated by a colon:
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply configuration/items.md]: Done

File: configuration/items.md [hunk 9]
```diff
@@ -296,8 +296,11 @@ In the example below, the "switch" icon has been selected:
 Switch Livingroom_Light "Livingroom Ceiling Light" <switch>
 ```
 
+#### Icons provided with openHAB
+
 openHAB provides a set of [classic icons](/docs/configuration/iconsets/classic/) by default.
-Users may add their own icons in either `png` or `svg` format in the openHAB icons configuration folder, `$OPENHAB_CONF/icons/classic/`.
+The community marketplace also provides other openHAB icon sets that can be installed, e.g. [this subset of iconify icons]( https://community.openhab.org/t/iconify-icon-provider/144508).
+Users may also add their own icons in either `png` or `svg` format in the openHAB icons configuration folder, `$OPENHAB_CONF/icons/classic/`.
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply configuration/items.md:303]: I kept only the last one.

File: configuration/items.md (hunk 3 above)
Comment (lolodomo) [reply configuration/items.md]: Fixed

File: configuration/items.md (hunk 4 above)
Comment (lolodomo) [reply configuration/items.md]: Done

File: configuration/items.md [hunk 10]
```diff
@@ -310,17 +313,67 @@ The following guidelines apply to user-added icon files:
...
+
+| Source name          | Source description                    | Main UI    | Basic UI                          | Android app       | iOS app           |
+| -------------------- | ------------------------------------- | ---------- | --------------------------------- | ----------------- | ----------------- |
+| `oh`                | Icons provided via the openHAB server | Supported  | Supported                         | Supported         | Supported         |
+| `material`         | Material icons                        | Supported  | Supported                         | Not yet supported | Not yet supported |
+| `f7`                | Framework7 icons                      | Supported  | Not yet supported                 | Not yet supported | Not yet supported |
+| `if` or `iconify` | iconify icons                         | Supported  | Supported but needs to be enabled | Not yet supported | Not yet supported |
+
+You can find a list of [all Material icons here](https://fonts.google.com/icons?icon.set=Material+Icons).
+
```
Comment (mueller-ma) [inline configuration/items.md:366]: You could move these links to a new coloumn in the table above. For source "oh" there's already a docs page that can be linked.

Comment (lolodomo) [reply configuration/items.md:366]: Or even by adding the link in the source description column.

**Comment by lolodomo** [comment]
> The case of dynamic icons and hyphens in icon name will be updated later in a separate PR in case a change is really implemented in the core framework.

As the PR for hyphen ( openhab/openhab-core#3548 ) is about to be merged, I am going to include the corresponding changes in this PR.

**Comment by stefan-hoehn** [comment]
@lolodomo Let me know / Ping me when you are done, so I can review it quickly and merged asap.

File: configuration/items.md (hunk 5 above)
Comment (lolodomo) [reply configuration/items.md:339]: Done

File: configuration/items.md (hunk 10 above)
Comment (lolodomo) [reply configuration/items.md]: Done

**Comment by lolodomo** [comment]
> @lolodomo Let me know / Ping me when you are done, so I can review it quickly and merged asap.

@stefan-hoehn : it is now finished and this updated page is now fully inline with what is available with the today OH snapshot.

**Review by stefan-hoehn — APPROVED** [review]

**Comment by stefan-hoehn** [comment]
great and comprehensive description of topic! LGTM.

**Pull request merged**